src/Symfony/Component/Workflow/TransitionBlockerList.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow;
+
+/**
+ * A list of transition blockers.
+ */
+class TransitionBlockerList implements \IteratorAggregate, \Countable, \ArrayAccess
```

Comment:
Do we really need the ArrayAccess ? I don't think so

Reply:
I agree ArrayAccess should be dropped and maybe even an interface introduced that only grants read access

Reply:
:-1: for the interface.

Reply:
I'll remove the `ArrayAccess`, but I found that useful to think of the instances of this class as "smarter arrays".

I'll also remove the `::remove()` method, since I didn't find it useful in any use case from the very beginning.

I disagree, that the thing should be completely immutable, though. The common pattern in Transition Guards is to start with an empty `TransitionBlockerList` and then `::add()` blockers as `if` statements evaluate, that the transition isn't possible.